cmd/kyverno/main.go

```diff
@@ -181,6 +184,14 @@ func createrLeaderControllers(
 	leaderControllers = append(leaderControllers, internal.NewController(exceptionWebhookControllerName, exceptionWebhookController, 1))
 
 	if generateVAPs {
+		// check if the controller has the required permissions to generate validating admission policies.
+		gvr := schema.GroupVersionResource{Group: "admissionregistration.k8s.io", Version: "v1alpha1", Resource: "validatingadmissionpolicies"}
+		vapPermissions := validatingadmissionpolicy.HasRequiredPermissions(gvr, checker)
+
+		// check if the controller has the required permissions to generate validating admission policy bindings.
+		gvr = schema.GroupVersionResource{Group: "admissionregistration.k8s.io", Version: "v1alpha1", Resource: "validatingadmissionpolicybindings"}
+		vapbindingPermissions := validatingadmissionpolicy.HasRequiredPermissions(gvr, checker)
+
```

Comment:
Suggest moving this to `pkg/controllers/validatingadmissionpolicy-generate/controller.go`.

Reply:
Done.